hkmc2/shared/src/test/mlscript-compile/Stack.mls

```diff
@@ -12,7 +12,7 @@ module Stack[T] with ...
 // TODO support type/module overloading
 type T[A]
 
-data class (::) Cons[A](head: A, tail: Stack[A])
+data class (::) Cons[A](head: A, tail: Cons[A] | Nil)
```

Comment:
I guess this change is to avoid the type parameter number check? How about
```suggestion
data class (::) Cons[A](head: A, tail: T[A])
```

Reply:
It works.